### PR TITLE
Use apidoc directive in stream/stream-substream.md (#22904)

### PR DESCRIPTION
**Pull request by Captain1653**
References #22904

### CODE REVIEW

File: akka-docs/src/main/paradox/stream/stream-substream.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ To use Akka Streams, add the module to your project:
 
 ## Introduction
 
-Substreams are represented as `SubSource` or `SubFlow` instances, on which you can multiplex a single `Source` or `Flow`
+Substreams are represented as `SubSource` or @apidoc[akka.stream.*.SubFlow] instances, on which you can multiplex a single @apidoc[akka.stream.*.Source] or @apidoc[akka.stream.*.Flow]
```
Comment (Captain1653) [inline akka-docs/src/main/paradox/stream/stream-substream.md:18]: There is not `scaladsl.SubSource` in the akka. Shoud I use here only 'javadsl.SubSource'?

Comment (raboof) [reply akka-docs/src/main/paradox/stream/stream-substream.md:18]: AFAICS we don't have `SubSource` in any of the public Scala API's, so perhaps the best thing would be to only show it when looking at the Java version of the docs?

File: akka-docs/src/main/paradox/stream/stream-substream.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ To use Akka Streams, add the module to your project:
 
 ## Introduction
 
-Substreams are represented as `SubSource` or `SubFlow` instances, on which you can multiplex a single `Source` or `Flow`
+Substreams are represented as @java[@javadoc[SubSource](akka.stream.javadsl.SubSource) or] @apidoc[akka.stream.*.SubFlow] instances, on which you can multiplex a single @apidoc[akka.stream.*.Source] or @apidoc[akka.stream.*.Flow]
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/stream-substream.md:18]:
```suggestion
Substreams are represented as @java[@javadoc[SubSource](akka.stream.javadsl.SubSource) or] @apidoc[akka.stream.*.SubFlow] instances, on which you can multiplex a single @java[@apidoc[akka.stream.*.Source] or] @apidoc[akka.stream.*.Flow]
```

**Review by raboof — APPROVED** [review]
LGTM with one suggestion

**Review by raboof — APPROVED** [review]

**Pull request merged**